From 19a656562dbb6f54c0c6d54d769b19602b0c37c1 Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Wed, 23 Jun 2021 13:35:50 +0900
Subject: [PATCH 04/10] [ozone/wayland] Apply default preedit style if empty
 style is passed

Without this fix, no style is applied on Omnibox while default style is
applied on contents area. This difference will confuses users.

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 .../wayland/host/wayland_input_method_context.cc    | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/ui/ozone/platform/wayland/host/wayland_input_method_context.cc b/ui/ozone/platform/wayland/host/wayland_input_method_context.cc
index 435f848649ee8..3beb6a3fa7d17 100644
--- a/ui/ozone/platform/wayland/host/wayland_input_method_context.cc
+++ b/ui/ozone/platform/wayland/host/wayland_input_method_context.cc
@@ -323,8 +323,20 @@ void WaylandInputMethodContext::OnPreeditString(
     base::StringPiece text,
     const std::vector<SpanStyle>& spans,
     int32_t preedit_cursor) {
+  VLOG(1) << "WaylandInputMethodContext::OnPreeditString: " << text;
+  VLOG(1) << "WaylandInputMethodContext::OnPreeditString: cursor: " << preedit_cursor;
+
   ui::CompositionText composition_text;
   composition_text.text = base::UTF8ToUTF16(text);
+
+  if (spans.empty()) {
+    VLOG(1) << "WaylandInputMethodContext::OnPreeditString: Style is empty, apply default one";
+    ImeTextSpan text_span;
+    text_span.thickness = ImeTextSpan::Thickness::kThick;
+    text_span.start_offset = 0;
+    text_span.end_offset = composition_text.text.length();
+    composition_text.ime_text_spans.push_back(std::move(text_span));
+  }
   for (const auto& span : spans) {
     ImeTextSpan text_span;
     auto start_offset = OffsetFromUTF8Offset(text, span.index);
@@ -357,6 +369,7 @@ void WaylandInputMethodContext::OnPreeditString(
 }
 
 void WaylandInputMethodContext::OnCommitString(base::StringPiece text) {
+  VLOG(1) << "WaylandInputMethodContext::OnCommitString: " << text;
   ime_delegate_->OnCommit(base::UTF8ToUTF16(text));
 }
 
-- 
2.34.1

